src/Execution/Arguments/ArgumentSet.php

```diff
@@ -61,19 +61,38 @@ public function has(string $key): bool
 
     /**
      * Add a value at the dot-separated path.
-     *
-     * Works just like @see \Illuminate\Support\Arr::add().
+     * Asterisks may be used to indicate wildcards.
      *
      * @param  mixed  $value  any value to inject
      */
-    public function addValue(string $path, $value): self
+    public function addValue(string $path, mixed $value): self
     {
         $argumentSet = $this;
         $keys = explode('.', $path);
 
+        self::applyValue($keys, $value, $argumentSet);
+
+        return $this;
+    }
+
+    /**
+     * @param  array<string>  $keys
+     * @param  mixed  $value any value to inject
+     * @param  ArgumentSet  $argumentSet
+     */
+    private static function applyValue(array $keys, $value, ArgumentSet $argumentSet): void
+    {
         while (count($keys) > 1) {
             $key = array_shift($keys);
 
+            if ($key === '*') {
+                foreach ($argumentSet as $argument) {
```

Comment:
Not sure if we should throw an error with it's not an array here?

Reply:
I've updated it now. Not sure if i used the correct exception.

Reply:
Use `DefinitionException`, in this case the error must be solved by the developer.